### PR TITLE
add org.openrgb.OpenRGB

### PR DESCRIPTION
**Pull request by abma**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.openrgb.OpenRGB.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+command: openrgb
+#rename-icon: springlobby
+#rename-desktop-file: springlobby.desktop
+#rename-appdata-file: springlobby.appdata.xml
+#finish-args:
+#- --share=network
+#- --socket=x11
+#- --share=ipc
+#- --device=dri
+#- --socket=pulseaudio
```
Comment (hfiguiere) [inline org.openrgb.OpenRGB.yaml:14]: without any finish args, it's unlikely the app will work.

File: org.openrgb.OpenRGB.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    url: https://github.com/ARMmbed/mbedtls.git
+    commit: 'f71e2878084126737cc39083e1e15afc459bd93d'
+- name: openrgb
+  buildsystem: simple
+  build-commands:
+   - qmake OpenRGB.pro LIBS+=-L/app/lib PREFIX=/app
+   - make install_target install_desktop install_pixmap
+  sources:
+  - type: git
+    url: https://github.com/abma/OpenRGB.git
```
Comment (hfiguiere) [inline org.openrgb.OpenRGB.yaml:44]: why do you use a fork?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.openrgb.OpenRGB.yaml (hunk 2 above)
Comment (abma) [reply org.openrgb.OpenRGB.yaml:44]: upstream some changes are needed: https://github.com/CalcProgrammer1/OpenRGB/pull/9

Comment (hfiguiere) [reply org.openrgb.OpenRGB.yaml:44]: usually we add patches

Comment (abma) [reply org.openrgb.OpenRGB.yaml:38]: atm i expect that the changes are (hopefully) accepted upstream: give it some time please. also i'm new to openrgb, so no clue what i missed.

File: org.openrgb.OpenRGB.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+app-id: org.openrgb.OpenRGB
+runtime:  org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: openrgb
+finish-args:
+- --socket=x11
```
Comment (hfiguiere) [inline org.openrgb.OpenRGB.yaml:7]:
```suggestion
- --socket=x11
- --share=ipc
```

this is required for X11.

No wayland support?

File: org.openrgb.OpenRGB.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+- name: mbedtls
+  buildsystem: simple
+  build-commands:
+   - make install DESTDIR=/app
+  sources:
+  - type: git
+    url: https://github.com/ARMmbed/mbedtls.git
+    commit: 'f71e2878084126737cc39083e1e15afc459bd93d'
+- name: openrgb
+  buildsystem: simple
```
Comment (hfiguiere) [inline org.openrgb.OpenRGB.yaml:32]: doesn't `buildsystem: qmake` work here?

File: org.openrgb.OpenRGB.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.openrgb.OpenRGB.yaml:38]: the removal of ` -lstdc++fs` unconditionally will break elsewhere

again the best practice is to use upstream code and then add patches on top as needed.

**Comment by hfiguiere** [comment]
also I can see you'll need to `rename-desktop-file`, `rename-icon`, and I don't even see an appstream file.

File: org.openrgb.OpenRGB.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.openrgb.OpenRGB.yaml:38]: what I mean here is that acceptance of this change as is by upstream is unlikely.

File: org.openrgb.OpenRGB.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- --socket=x11
+- --device=all
+
+modules:
+- name: libusb
+  buildsystem: autotools
+  sources:
+  - type: git
+    url: 'https://github.com/libusb/libusb.git'
+    commit: '1a906274a66dd58bf81836db1306902d4a7dc185'
```
Comment (gasinvein) [inline org.openrgb.OpenRGB.yaml:16]: Any reason to use unreleased git snapshots?

File: org.openrgb.OpenRGB.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- name: libhdapi
+  buildsystem: cmake-ninja
+  sources:
+   - type: git
+     url: https://github.com/libusb/hidapi.git
+     commit: 'e9f8c3e75e9891de159aeb18fee1685637c68e84'
```
Comment (gasinvein) [inline org.openrgb.OpenRGB.yaml:22]: Same here, why not a released version?

File: org.openrgb.OpenRGB.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+- name: mbedtls
+  buildsystem: simple
+  build-commands:
+   - make install DESTDIR=/app
+  sources:
+  - type: git
+    url: https://github.com/ARMmbed/mbedtls.git
+    commit: 'f71e2878084126737cc39083e1e15afc459bd93d'
```
Comment (gasinvein) [inline org.openrgb.OpenRGB.yaml:30]: Same here, why not a released version?

**Review by gasinvein — COMMENTED** [review]
Please conform to [the YAML style guide](https://github.com/flathub/flathub/wiki/YAML-Style-Guide). It's 2 whitespace before array item and 1 empty line between modules.

File: org.openrgb.OpenRGB.yaml (hunk 2 above)
Comment (abma) [reply org.openrgb.OpenRGB.yaml]: the changes where merged upstream, so changed the url to upstream.

File: org.openrgb.OpenRGB.yaml (hunk 5 above)
Comment (hfiguiere) [reply org.openrgb.OpenRGB.yaml:18]: we have a shared-module for this.

**Comment by hfiguiere** [comment]
bot, build org.openrgb.OpenRGB

File: org.openrgb.OpenRGB.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+- name: openrgb
+  buildsystem: simple
+  build-commands:
+   - qmake OpenRGB.pro LIBS+=-L/app/lib PREFIX=/app
+   - make install_target install_desktop install_pixmap
+  sources:
+  - type: git
+    url: https://gitlab.com/OpenRGB/OpenRGB.git
+    commit: 'b8246b614016cf83b758ddf006c3f436e929b968'
```
Comment (hfiguiere) [inline org.openrgb.OpenRGB.yaml:41]: it seems that this commit doesn't exist upstream.

**Comment by barthalion** [comment]
bot, build org.openrgb.OpenRGB

**Comment by hfiguiere** [comment]
there is not appdata file to be found.

**Comment by hfiguiere** [comment]
and i suspect no .desktop file either.

**Comment by 1player** [comment]
Any update on this? I might look into it myself (within the next month) if this PR has been abandoned.

**Comment by hfiguiere** [comment]
Superseded by #2677

**Pull request closed without merging**